server/main_server/src/agent_main.cpp

```diff
@@ -0,0 +1,1326 @@
+
+    auto deinit_shared_memory_for_plugin(const nlohmann::json& _plugin_object) -> bool
+    {
+        const auto itr = _plugin_object.find(irods::KW_CFG_SHARED_MEMORY_INSTANCE);
+
+        if (_plugin_object.end() != itr) {
+            const auto& mem_name = itr->get_ref<const std::string&>();
+            removeSharedMemory(mem_name);
+            resetMutex(mem_name.c_str());
+            return true;
+        }
+
+        return false;
+    } // deinit_shared_memory_for_plugin
+
+    auto deinit_shared_memory_for_plugins() -> irods::error
+    {
+        try {
+            const auto config_handle{irods::server_properties::instance().map()};
+            const auto& config{config_handle.get_json()};
+
+            for (const auto& item : config.at(irods::KW_CFG_PLUGIN_CONFIGURATION).items()) {
+                for (const auto& plugin : item.value().items()) {
+                    deinit_shared_memory_for_plugin(plugin.value());
+                }
+            }
+        }
+        catch (const irods::exception& e) {
+            return irods::error(e);
+        }
+        catch (const std::exception& e) {
+            return ERROR(SYS_INTERNAL_ERR, e.what());
+        }
+
+        return SUCCESS();
+    } // deinit_shared_memory_for_plugins
+
+    // TODO Remove this since initAgent() covers this for agents.
+#if 0
+    auto initServer(RsComm& _comm) -> int
+    {
+        if (const auto ec = initServerInfo(0, &_comm); ec < 0) {
+            log_af::info("{}: initServerInfo error, status = {}", __func__, ec);
+            return ec;
+        }
+
+        // TODO Unnecessary?
+        //resc_mgr.print_local_resources();
+
+        printZoneInfo();
+
+        rodsServerHost_t* rodsServerHost{};
+        if (const auto ec = getRcatHost(PRIMARY_RCAT, nullptr, &rodsServerHost); ec < 0 || !rodsServerHost) {
+            return ec;
+        }
+
+        std::string svc_role;
+        if (const auto res = get_catalog_service_role(svc_role); !res.ok()) {
+            log_af::error("{}: Could not get server role. [error code={}]", __func__, res.code());
+            return res.code();
+        }
+
+        if (LOCAL_HOST == rodsServerHost->localFlag) {
+            if (irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role) {
+                disconnectRcat();
+            }
+        }
+        else if (rodsServerHost->conn) {
+            rcDisconnect(rodsServerHost->conn);
+            rodsServerHost->conn = nullptr;
+        }
+
+        return 0;
+    } // initServer
+#endif
+
+    auto initServerMain(RsComm& _comm) -> int
+    {
+        std::memset(&_comm, 0, sizeof(RsComm));
+        int status = getRodsEnv(&_comm.myEnv);
+        if (status < 0) {
+            log_af::error("{}: getRodsEnv error. status = {}", __func__, status);
+            return status;
+        }
+
+        // TODO Verify this is reading from the correct location.
+        // Consider non-pkg installs and pkg installs.
+        setRsCommFromRodsEnv(&_comm);
```

Comment:
This function reads data from the rodsEnv object embedded in the RsComm. Remove this TODO.

Reply:
Replaced this function with its implementation since the function is only used in one place and its name didn't align with the behavior.

The function has been removed.